api/service.go

```diff
@@ -0,0 +1,409 @@
+
+	err = json.Unmarshal(r, &rates)
+	if err != nil {
+		return
+	}
+	return
+}
+
+// GetExchanges fetches the current known state of all exchanges
+func (s *Service) GetExchanges() (state *ExchangeState, err error) {
+	r, err := s.client.Do("GET", "api/exchanges", "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &state)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (s *Service) GetTicketFeeRateSummary() (ticketInfo *apiTypes.MempoolTicketFeeInfo, err error) {
+	r, err := s.client.Do("GET", "api/mempool/sstx", "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &ticketInfo)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (s *Service) GetTicketFeeRate() (ticketFeeRate *apiTypes.MempoolTicketFees, err error) {
+	r, err := s.client.Do("GET", "api/mempool/sstx/fees", "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &ticketFeeRate)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (s *Service) GetNHighestTicketFeeRate(nHighest int) (ticketFeeRate *apiTypes.MempoolTicketFees, err error) {
+	r, err := s.client.Do("GET", "api/mempool/sstx/fees/"+strconv.Itoa(nHighest), "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &ticketFeeRate)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (s *Service) GetTicketDetails() (ticketDetails *apiTypes.MempoolTicketDetails, err error) {
+	r, err := s.client.Do("GET", "api/mempool/sstx/details", "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &ticketDetails)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (s *Service) GetNHighestTicketDetails(nHighest int) (ticketDetails *apiTypes.MempoolTicketDetails, err error) {
+	r, err := s.client.Do("GET", "api/mempool/sstx/details/"+strconv.Itoa(nHighest), "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &ticketDetails)
+	if err != nil {
+		return
+	}
+	return
+}
+
+// GetAddress returns the balances and transactions of an address.
+// The returned transactions are sorted by block height, newest blocks first.
+func (s *Service) GetAddress(address string) (addressState *AddressState, err error) {
+	blockBookBaseUrl := blockbookMainnet
+	if s.chainParams.Name == chaincfg.TestNet3Params().Name {
+		blockBookBaseUrl = blockbookTestnet
+	}
+
+	r, err := s.client.Do("GET", blockBookBaseUrl+"api/v2/address/"+address, "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &addressState)
+	if err != nil {
+		return
+	}
+	return
+}
+
+// GetXpub Returns balances and transactions of an xpub.
+func (s *Service) GetXpub(xPub string) (xPubBalAndTxs *XpubBalAndTxs, err error) {
+	blockBookBaseUrl := blockbookMainnet
+	if s.chainParams.Name == chaincfg.TestNet3Params().Name {
+		blockBookBaseUrl = blockbookTestnet
+	}
+
+	r, err := s.client.Do("GET", blockBookBaseUrl+"api/v2/xpub/"+xPub, "")
+	if err != nil {
+		return
+	}
+
+	err = json.Unmarshal(r, &xPubBalAndTxs)
+	if err != nil {
+		return
+	}
+	return
+}
+
+// GetTicker returns market ticker data for the supported exchanges.
```

Comment:
This comment can be made more elaborate, for example
1. specifying that Bittrex doesn't support testnet.
2. from the supported function list, it says `The market pair should be in the format: DRC-BTC.`, but for Binance i get  `error:400 Bad Request:'{"code":-1121,"msg":"Invalid symbol."}'`, perhaps the market formats for the diffrent exchanges can be shown
3. For i get `&{KuCoin DRC-BTC 0 0 0} <nil>`, using `DCR-BTC` market pair, the LastTradeRate is not returned

Reply:
might be that DCR-BTC Isn't supported on kucoin

Reply:
> might be that DCR-BTC Isn't supported on kucoin

DCR_BTC is supported on kucoin. You are running on testnet. They support other coins on testnet but they don't support dcr-btc.

Reply:
> 

You are getting this errors because you are on testnet3. These exchanges doesn't support dcr-btc on testnet.